velox/type/ShortDecimal.h

```diff
@@ -23,11 +23,14 @@
 
 namespace facebook::velox {
 
+struct LongDecimal;
+
 struct ShortDecimal {
  public:
   // Default required for creating vector with NULL values.
   ShortDecimal() = default;
-  constexpr explicit ShortDecimal(int64_t value) : unscaledValue_(value) {}
+  constexpr ShortDecimal(int64_t value) : unscaledValue_(value) {}
+  ShortDecimal(LongDecimal& value);
```

Comment:
What is this constructor for? Is it valid to create a short decimal value from a long decimal value?

Reply:
This is required for the sum aggregate template setup. The initialization `0` is a LongDecimal which gets passed as a ShortDecimal value. I added a check to ensure validity.

Reply:
We can specialize there if needed. But I thought this won't hurt as long as there is a validity check.

Reply:
Is it this code?

`*exec::Aggregate::value<ResultType>(groups[i]) = 0;`

Can it be changed to ` = ResultType(0)`?

Reply:
The issue is with the template parameter types
https://github.com/facebookincubator/velox/blob/717c5a97515e8ad410e6c9681be7add73f48925f/velox/functions/prestosql/aggregates/SimpleNumericAggregate.h#L130-L151
`TData initialValue` is `LongDecimal` type which gets passed to `updateNonNullValue(char* group, TInput value, Update updateValue)` where `TInput` is `ShortDecimal` type.